caustic/parametrized.py

```diff
@@ -429,72 +312,59 @@ def __getattribute__(self, key):
                 raise e
 
     @property
-    def params(self) -> dict[str, OrderedDict[str, Parameter]]:
-        """
-        Gets all of the static and dynamic Params for this component and its descendants
-        in the correct order.
-
-        Returns:
-            dict[str, OrderedDict[str, Parameter]]: Dictionary containing all static and dynamic parameters.
-        """
-        static = OrderedDict()
-        dynamic = OrderedDict()
+    def module_params(self) -> NestedNamespaceDict:
+        static = NestedNamespaceDict()
+        dynamic = NestedNamespaceDict()
         for name, param in self._params.items():
             if param.static:
                 static[name] = param
             else:
                 dynamic[name] = param
-
-        return {"static": static, "dynamic": dynamic}
-
+        return NestedNamespaceDict([("static", static), ("dynamic", dynamic)])
+    
     @property
-    def static_params(self) -> OrderedDict[str, Parameter]:
-        """
-        Returns all the static parameters of the object.
-
-        Returns:
-            OrderedDict[str, Parameter]: An ordered dictionary of static parameters.
-        """
-        return self.params["static"]
+    def params(self) -> NestedNamespaceDict:
+        static = NestedNamespaceDict()
+        dynamic = NestedNamespaceDict()
+        def _get_params(module):
+            if module._childs != {}:
```

Comment:
maybe `if len(module._childs) > 0`, its more general

Reply:
Now that I think about, `if module._childs` is even more general